guides/source/active_model_basics.md

```diff
@@ -285,101 +800,308 @@ irb> person = Person.new
 
-`ActiveModel::Model` allows implementing models similar to `ActiveRecord::Base`.
+```irb
+irb> Person.model_name.singular
+=> "person"
+```
 
-```ruby
-class EmailContact
-  include ActiveModel::Model
+**`plural`** returns the plural class name of a record or class.
 
-  attr_accessor :name, :email, :message
-  validates :name, :email, :message, presence: true
+```irb
+irb> Person.model_name.plural
+=> "people"
+```
 
-  def deliver
-    if valid?
-      # deliver email
-    end
-  end
-end
+**`element`** returns a potentially namespaced class name to snake_case.
+
+```irb
+irb> Person.model_name.element
+=> "person"
+```
+
+**`human`** transforms the model name into a more human format, using I18n. By
+default, it will underscore and then humanize the class name.
+
+```irb
+irb> Person.model_name.human
+=> "Person"
+```
+**`collection`** returns the name of a table like Rails does for models to table
+names. It uses the pluralize method on the last word in the string.
+
+```irb
+irb> Person.model_name.collection
+=> "people"
+```
+
+**`param_key`** returns a string to use for params names. It differs for
+namespaced models regarding whether it's inside an isolated engine.
+
+```irb
+irb> Person.model_name.param_key
+=> "person"
+```
+
+**`i18n_key`** returns the name of the i18n key. It underscores the model name
+and then returns it as a symbol.
+
+```irb
+irb> Person.model_name.i18n_key
+=> :person
 ```
 
-When including `ActiveModel::Model` you get all the features from `ActiveModel::API`.
+**`route_key`** returns a string to use while generating route names. It differs
+for namespaced models regarding whether it's inside isolated engine.
+
+```irb
+irb> Person.model_name.route_key
+=> "people"
+```
+
+**`singular_route_key`** returns a string to use while generating route names.
+It differs for namespaced models regarding whether it's inside isolated engine.
+
+```irb
+irb> Person.model_name.singular_route_key
+=> "person"
+```
+
+**`uncountable?`** identifies whether the class name of a record or class is
+uncountable.
+
+```irb
+irb> Person.model_name.uncountable?
+=> false
+```
+
+### Model
+
+`ActiveModel::Model` includes ActiveModel::API for the required interface to
+allow an object to interact with Action Pack and Action View, but it will be
+extended in the future to add more functionality.
+
+Currently, when including `ActiveModel::Model` you get the features from
+[ActiveModel::API](active_model_basics.html#api).
 
 ### Serialization
 
-`ActiveModel::Serialization` provides basic serialization for your object.
-You need to declare an attributes Hash which contains the attributes you want to
-serialize. Attributes must be strings, not symbols.
+`ActiveModel::Serialization` provides basic serialization to a serializable hash
+for your objects. You need to declare an attributes Hash which should contain
+the attributes you want to serialize. Attributes must be strings, not symbols.
 
```

Comment:
I think between the 2 opening paragraphs here, the previous one might suffice. And hash doesn't need capitalization.